### DISCUSSION TITLE
Bundled Web UI for ESP32 is now optional

### DISCUSSION THREAD

**Discussion by thebentern** [meshtastic/firmware#5381] [Announcements]
# We are making the bundled Web UI on ESP32 optional 
The initial 2.5.13 release (2.5.13.295278b) of the Meshtastic Firmware came with some [issues](https://github.com/meshtastic/firmware/issues/5358) on ESP32 devices on a clean installation. If you noticed a boot log like this, you likely experienced the problem:

`DEBUG | ??:??:?? 0 Filesystem files (1040384/1048576 Bytes):`

The firmware is setup by default to pull the latest tagged release from the `meshtastic/web` repo and untar the contents into a static directory and build the LittleFS filesystem binary for ESP32 devices.  The initial 2.5.13 release used this process, however there were some problems with the latest Web UI size and packaging. This caused the filesystem binary to be so large in flash utilization that even adding a new node to the device's  NodeDB would cause a panic (crash) when the subsequent attempt to save the changes to the flash occurred.  We pulled this initial release shortly after some of the problem reports came in. 

### Here's how we're addressing the issue going forward avoid encountering the problem again:

1. Because the overwhelming majority of users rely on the apps and Bluetooth connection, we are building the default littlefs binary _without_ the WebUI, freeing up more space.
![image](https://github.com/user-attachments/assets/644aa37c-7a7c-4805-ad1e-a9e3d180ef18)

> First boot filesystem _without_ Web UI:
`DEBUG | ... Filesystem files (8192/1048576 Bytes):`

>First boot filesystem _with_ Web UI:
`DEBUG | ... Filesystem files (897024/1048576 Bytes):`

3. We have also modified our build process to pull from a specific tag rather than simply `latest`, which has made the firmware release non-deterministic. We don't like these sorts of surprises, and we know you don't either. The recut 2.5.13 firmware release is pulling from the previous web tag which does not have the aforementioned issues.
4. We have added some logic into the bootup sequence to attempt to clear out the offending releases `/static/static` assets from the file system to hotfix any devices in the wild with this issue.

We believe these changes will increase the reliability of the firmware releases on the ESP32 platform and align better with how most users utilize their devices. Additionally, ESP32 devices are not the most reliable host for a web application, especially when combined with the resource usage the Meshtastic firmware already demands. It is recommended that for more reliable operation, you should host the web client application separately on your network.

**Comment by shurillu** [comment]
Hello, I'm building a custom ESP32 based hardware and so I write down a custom variant file for that, compiling and uploading the firmware with VisualStudio Code.
It work without a problem, but I can't find a way to host the web client on the ESP32 (I need it because it's a fixed node on my roof and I want to connect to it with WiFi).
There is a way to configure the project to include the web UI in the compilation process?

Thanks, regards.

Stefano

**Reply by CozeDoge** [reply]
Hi Shurillu, did you ever figure your question out?
I'm in the same position and can get the web-server enabled, but no UI.
